src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -1,78 +1,77 @@
-import argparse
-import os
 import re
 from pathlib import Path
 
 import pytest
 
+from diffpy.labpdfproc.labpdfprocapp import get_args
 from diffpy.labpdfproc.tools import known_sources, set_output_directory, set_wavelength
 
 params1 = [
-    ([None], ["."]),
-    (["."], ["."]),
-    (["new_dir"], ["new_dir"]),
-    (["existing_dir"], ["existing_dir"]),
+    ([], ["."]),
+    (["--output-directory", "."], ["."]),
+    (["--output-directory", "new_dir"], ["new_dir"]),
+    (["--output-directory", "input_dir"], ["input_dir"]),
 ]
 
 
 @pytest.mark.parametrize("inputs, expected", params1)
-def test_set_output_directory(inputs, expected, tmp_path):
-    directory = Path(tmp_path)
-    os.chdir(directory)
+def test_set_output_directory(inputs, expected, user_filesystem):
+    tmp_dir = user_filesystem
+    expected_output_directory = tmp_dir / expected[0]
```

Comment:
change to something like `expected_output_directory = user_filesystem / expected[0]` (I am not sure if we need to use `Path(user_filesystem)` or sthg.... as I am not sure what the fixture returns.....bbut you can figure it out.  This works what you are doing, but again, I am trying to make the code more readable and understandable in the future